posthog/temporal/batch_exports/http_batch_export.py

```diff
@@ -0,0 +1,251 @@
+    CreateBatchExportRunInputs,
+    UpdateBatchExportRunStatusInputs,
+    create_export_run,
+    execute_batch_export_insert_activity,
+    get_data_interval,
+    get_rows_count,
+    iter_records,
+    json_dumps_bytes,
+)
+from posthog.temporal.batch_exports.clickhouse import get_client
+from posthog.temporal.batch_exports.metrics import (
+    get_bytes_exported_metric,
+    get_rows_exported_metric,
+)
+from posthog.temporal.common.logger import bind_temporal_worker_logger
+
+
+def http_default_fields() -> list[BatchExportField]:
+    return [
+        BatchExportField(expression="toString(uuid)", alias="uuid"),
+        BatchExportField(expression="timestamp", alias="timestamp"),
+        BatchExportField(expression="event", alias="event"),
+        BatchExportField(expression="nullIf(properties, '')", alias="properties"),
+        BatchExportField(expression="toString(distinct_id)", alias="distinct_id"),
+        BatchExportField(expression="toJSONString(elements_chain)", alias="elements_chain"),
+    ]
+
+
+@dataclass
+class HttpInsertInputs:
+    """Inputs for HTTP insert activity."""
+
+    team_id: int
+    url: str
+    token: str
+    data_interval_start: str
+    data_interval_end: str
+    exclude_events: list[str] | None = None
+    include_events: list[str] | None = None
+    batch_export_schema: BatchExportSchema | None = None
+
+
+async def post_json_file_to_url(url, batch_file):
+    batch_file.seek(0)
+    async with aiohttp.ClientSession() as session:
+        headers = {"Content-Type": "application/json"}
+        async with session.post(url, data=batch_file, headers=headers, raise_for_status=True) as response:
+            return response
+
+
+@activity.defn
+async def insert_into_http_activity(inputs: HttpInsertInputs):
+    """Activity streams data from ClickHouse to an HTTP Endpoint."""
+    logger = await bind_temporal_worker_logger(team_id=inputs.team_id, destination="HTTP")
+    logger.info(
+        "Exporting batch %s - %s",
+        inputs.data_interval_start,
+        inputs.data_interval_end,
+    )
+
+    async with get_client() as client:
+        if not await client.is_alive():
+            raise ConnectionError("Cannot establish connection to ClickHouse")
+
+        count = await get_rows_count(
+            client=client,
+            team_id=inputs.team_id,
+            interval_start=inputs.data_interval_start,
+            interval_end=inputs.data_interval_end,
+            exclude_events=inputs.exclude_events,
+            include_events=inputs.include_events,
+        )
+
+        if count == 0:
+            logger.info(
+                "Nothing to export in batch %s - %s",
+                inputs.data_interval_start,
+                inputs.data_interval_end,
+            )
+            return
+
+        logger.info("BatchExporting %s rows", count)
+
+        if inputs.batch_export_schema is not None:
+            raise NotImplementedError("Batch export schema is not supported for HTTP export")
```

Comment:
I left it in the inputs to make this easier to enable later. I could also remove it from all the inputs I guess.

Reply:
Right, because the only goal right now is migrations that are only available to us.